### PR TITLE
[Lake] Add basic TabletWriter interfaces and default implementation

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

File: be/src/storage/lake/general_tablet_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+void GeneralTabletWriter::close() {
+    if (!_finished) {
+        // Delete files
+        auto group = _tablet.group();
+        auto maybe_fs = FileSystem::CreateSharedFromString(group);
+        if (!maybe_fs.ok()) {
+            return;
+        }
+        auto fs = std::move(maybe_fs).value();
+        for (auto&& name : _files) {
```
Comment (imay) [inline be/src/storage/lake/general_tablet_writer.cpp:55]: why use rvalue reference.

File: be/src/storage/lake/general_tablet_writer.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Status GeneralTabletWriter::finish() {
+    RETURN_IF_ERROR(flush_segment_writer());
+    _finished = true;
+    return Status::OK();
+}
+
+void GeneralTabletWriter::close() {
+    if (!_finished) {
+        // Delete files
+        auto group = _tablet.group();
```
Comment (imay) [inline be/src/storage/lake/general_tablet_writer.cpp:49]: Why name it group?

File: be/src/storage/lake/general_tablet_writer.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/lake/general_tablet_writer.cpp:55]: This is not an rvalue reference but a universal reference or, forwarding reference.

File: be/src/storage/lake/general_tablet_writer.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/storage/lake/general_tablet_writer.cpp:49]: This name comes from our design doc, a group is a common path prefix within object storage. 
Tablets with the same group prefix can share segment files with each other, so a group is the scan prefix to do garbage collection.
I think this is not a good name, but I haven't come up with another good enough name. When I think of a better name, I'll change it here

File: be/src/storage/lake/general_tablet_writer.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/lake/general_tablet_writer.cpp]: I replaced `auto&&` with `const auto&` for easier understanding，PTAL.

**Review by imay — DISMISSED** [review]

File: be/src/storage/lake/general_tablet_writer.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Status GeneralTabletWriter::finish() {
+    RETURN_IF_ERROR(flush_segment_writer());
+    _finished = true;
+    return Status::OK();
+}
+
+void GeneralTabletWriter::close() {
+    if (!_finished) {
+        // Delete files
```
Comment (wyb) [inline be/src/storage/lake/general_tablet_writer.cpp:48]: Better to check `_files` empty first.

File: be/src/storage/CMakeLists.txt [hunk 4]
```diff
@@ -157,4 +157,5 @@ add_library(Storage STATIC
     cluster_id_mgr.cpp
     lake/tablet.cpp
     lake/tablet_manager.cpp
+    lake/general_tablet_writer.cpp
```
Comment (wyb) [inline be/src/storage/CMakeLists.txt:160]:
```suggestion
    lake/general_tablet_writer.cpp
    lake/tablet.cpp
    lake/tablet_manager.cpp
```

File: be/src/storage/lake/general_tablet_writer.cpp [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        auto group = _tablet.group();
+        auto maybe_fs = FileSystem::CreateSharedFromString(group);
+        if (!maybe_fs.ok()) {
+            return;
+        }
+        auto fs = std::move(maybe_fs).value();
+        for (const auto& name : _files) {
+            auto path = fmt::format("{}/{}", group, name);
+            (void)fs->delete_file(path);
```
Comment (wuleistarrocks) [inline be/src/storage/lake/general_tablet_writer.cpp:57]: Why delete segment files?

File: be/src/storage/lake/general_tablet_writer.cpp (hunk 3 above)
Comment (sduzh) [reply be/src/storage/lake/general_tablet_writer.cpp:48]: updated

File: be/src/storage/lake/general_tablet_writer.cpp (hunk 5 above)
Comment (sduzh) [reply be/src/storage/lake/general_tablet_writer.cpp]: `!_finished` means this is a failed or canceled task.

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_clang-format

**Comment by sduzh** [comment]
run starrocks_clang-tidy

**Review by chaoyli — APPROVED** [review]

**Pull request merged**